eclass/kernel-install.eclass

```diff
@@ -442,6 +595,28 @@ kernel-install_pkg_preinst() {
 	fi
 }
 
+# @FUNCTION: kernel-install_extract_from_uki
+# @USAGE: <initrd|linux> <kernel image dir>
+# @DESCRIPTION:
+# Extracts kernel image or initrd from UKI in the given directory
+kernel-install_extract_from_uki() {
+	[[ ${#} -eq 2 ]] || die "${FUNCNAME}: invalid arguments"
+	local extract_type=${1}
+	local image_path=${2}
+	local uki=${image_path}/uki.efi
+	local out_file=
+	if [[ ${extract_type} == linux ]]; then
+		out=$(dist-kernel_get_image_path)
```

Comment:
That's the full path. Also, aren't you passing it as `${image_path}` already?

Reply:
> Also, aren't you passing it as `${image_path}` already?

No I changed it to take the path as argument not the image itself